eos-updater-flatpak-installer/main.c

```diff
@@ -544,6 +594,91 @@ parse_installer_mode (const gchar              *mode,
   return TRUE;
 }
 
+static const gchar *
+format_remote_ref_action_type (EosUpdaterUtilFlatpakRemoteRefActionType    action_type,
+                               GError                                    **error)
+{
+  GEnumClass *enum_class = g_type_class_ref (EOS_TYPE_UPDATER_UTIL_FLATPAK_REMOTE_REF_ACTION_TYPE);
+  GEnumValue *enum_value = g_enum_get_value (enum_class, action_type);
+
+  g_type_class_unref (enum_class);
+
+  if (!enum_value)
+    {
+      g_set_error (error,
+                   G_IO_ERROR,
+                   G_IO_ERROR_FAILED,
+                   "Invalid action type %d",
+                   action_type);
+      return NULL;
+    }
+
+  return enum_value->value_nick;
+}
+
+static gboolean
+log_all_flatpak_ref_actions (const gchar  *title,
+                             GHashTable   *flatpak_ref_actions_for_this_boot,
+                             GError      **error)
+{
+  gpointer key, value;
+  GHashTableIter iter;
+
+  g_message ("%s:", title);
+
+  g_hash_table_iter_init (&iter, flatpak_ref_actions_for_this_boot);
+
+  while (g_hash_table_iter_next (&iter, &key, &value))
+    {
+      const gchar *source = (const gchar *) key;
+      GPtrArray *actions = (GPtrArray *) value;
+
+      gsize i;
+
+      g_message("  %s:", source);
+
+      for (i = 0; i < actions->len; ++i)
+        {
+          FlatpakRemoteRefAction *action = g_ptr_array_index (actions, i);
+          const gchar *formatted_action_type = NULL;
+          g_autofree gchar *formatted_ref = NULL;
+
+          formatted_action_type = format_remote_ref_action_type (action->type, error);
+
+          if (!formatted_action_type)
+            return FALSE;
+
+          formatted_ref = flatpak_ref_format_ref (FLATPAK_REF (action->ref));
+
+          g_message("    - %s %s:%s",
+                    formatted_action_type,
+                    flatpak_remote_ref_get_remote_name (action->ref),
+                    formatted_ref);
+        }
+    }
+
+  return TRUE;
+}
+
+static void
+log_all_flatpak_ref_action_progresses (GHashTable   *flatpak_ref_action_progresses)
+{
+  gpointer key, value;
+  GHashTableIter iter;
+
+  g_message ("Action application progresses:");
+
+  g_hash_table_iter_init (&iter, flatpak_ref_action_progresses);
+
+  while (g_hash_table_iter_next (&iter, &key, &value))
+    {
+      const gchar *source = (const gchar *) key;
+      gint64 progress = GPOINTER_TO_INT (value);
```

Comment:
You can’t safely store a 64-bit integer in a pointer, since we have to support 32-bit platforms.